internal/provider/resource_gitlab_project_membership.go

```diff
@@ -48,6 +48,12 @@ var _ = registerResource("gitlab_project_membership", func() *schema.Resource {
 				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice(validProjectAccessLevelNames, false)),
 				Required:         true,
 			},
+			"expires_at": {
+				Description:  "Expiration date for the project membership. Format: `YYYY-MM-DD`",
+				Type:         schema.TypeString, // Format YYYY-MM-DD
```

Comment:
NIT: since the `Description` already contains that information close the the type definition I think it's okay to omit that comment:

```suggestion
				Type:         schema.TypeString,
```